automated-testing/performance-testing/load-testing.md

```diff
@@ -0,0 +1,69 @@
+
+Additionally the results of a load test can also be used as data to help with capacity planning and calculating scalability. 
+
+## Load Testing Design Blocks
+
+There are a number of basic component which are required to carry out a load test.  
+
+1. In order to have meaningful results the system needs to be tested in a production-like environment with a network and hardware which closely resembles the expected deployment environment.
+
+2. The load will consist consist a module which simulates user activity. Of course what the composition of this "user activity" will be vary based on the type of application being tested. For example an e-commerce web site might simulate user browsing and purchasing items, but an IoT data ingestion pipeline would simulate a stream of device readings.
+
+3. The load test will be initiated from a component external to the target system which can control the amount of load applied. This can be a single agent, but may need to scaled to multiple slaves in order to achieve higher levels of activity. 
+
+4. Although not required to run a load test, it is advisable to have monitoring and/or logging in place to be able to measure the impact of the test and discover potential bottlenecks.
+
+## Applying the Load Testing
+
+### Planning
+
+1. **Identify key scenarios to measure**  - Gather these scenarios from Product Owner, they should provide a representative sample of real world traffic.
+2. **Determine expected normal and peak load for the scenarios** -  Determine a load level such as concurrent users or requests per second to find the size of the load test you will run.
+3. **Identify success criteria metrics** - These may be testing side such as response time and error rate, or it may be on the system side such as CPU and memory usage.
+
+### Execution
+
+It is recommended to use an existing testing framework (see below). These tools will provide a method of both specifying the user activity scenarios as well as how to execute those at load. It is common to slowly ramp up to your desired load just to be safe. Once you have reached your defined workload. 
+
+Before starting please be aware of any restrictions on your network such as DDOS protection where you may need to notify a network administrator or apply for an exemption.
+
+### Further Testing
+After completing your load test you should be set up to continue on to additional related testing such as;
+- **Soak Testing** - Performing a load test over an extended period of time to ensure long term stability.
+- **Stress Testing** - Gradually increasing the load to find the limits of the system and identify the maximum capacity.
+- **Spike Testing** - Introduce a sharp short-term increase into the load scenarios.
+- **Scalability Testing** - Re-testing of a system as your expand horizontally or vertically to measure how it scales.

```

Comment:
Should the descriptions be added under the performance testing page, and then linked from here?

Reply:
I thought about this, but i think it is meaningful here since they are specifically extensions of a load test 
however we can maybe reconsider this if we restructure or add additional sub pages.